src/esbuild/external.ts

```diff
@@ -28,6 +28,9 @@ export const externalPlugin = ({
           if (match(args.path, resolvePatterns)) {
             return
           }
+          if (match(args.path, noExternal)) {
```

Comment:
The same logic repeats in second onResolve callback: https://github.com/egoist/tsup/blob/d074dd9c33adf85bcd149e79c9d141517a956ce0/src/esbuild/external.ts#L41 Looks like branching shall be implemented inside single callback:
```ts
match(args.path, noExternal)
if (skipNodeModulesBundle) {
  match(args.path, resolvePatterns)
  NON_NODE_MODULE_RE.test(args.path)
}
match(args.path, external)